src/analysis/souffle/tests/arcs_fact_tests/fact_test_driver.cc

```diff
@@ -26,28 +26,54 @@
 
 #include "souffle/SouffleInterface.h"
 
-int main(int argc, char **argv) {
+int run_test(std::string const &test_name) {
   // We want one command line arg, the name of the current test module.
-  assert(argc == 2);
-  std::string const test_name = std::string(argv[1]);
-
   std::unique_ptr<souffle::SouffleProgram> prog(
       souffle::ProgramFactory::newInstance(test_name));
   assert(prog != nullptr);
 
   prog->run();
 
-  souffle::Relation *test_passes =
-    prog->getRelation("testPasses");
+  souffle::Relation *test_failures = prog->getRelation("testFails");
+  souffle::Relation *all_tests = prog->getRelation("allTests");
+
+  assert(test_failures != nullptr);
 
-  int const test_passes_size = test_passes->size();
-  assert((test_passes_size == 0) || (test_passes_size == 1));
-  std::cout << "Test " << test_name;
-  if (test_passes_size == 1) {
-    std::cout << " passes." << std::endl;
+  bool const test_is_trivial = all_tests->size() == 0;
+  bool const test_has_failures = test_failures->size() > 0;
+
+  if (test_is_trivial) {
+    std::cout
+      << "Test "
+      << test_name
+      << " does not have any test conditions."
+      << std::endl;
+    prog->printAll();
+    return 1;
+  } else if (test_has_failures) {
+    std::cout << "Test " << test_name << " failed." << std::endl;
+    prog->printAll();
```

Comment:
Does this print the entire program? It would be more useful to just print the failing test names. We can do it in a separate PR.

Reply:
It doesn't print the entire program, it prints all of the output rules in the program. At this time, this includes only the test names and the failing tests, both of which are important information.